[AOTI] add flag AOT_INDUCTOR_ENABLE_LTO

helloguo opened the pull request:
Add env var AOT_INDUCTOR_ENABLE_LTO to enable clang's ThinLTO by setting AOT_INDUCTOR_ENABLE_LTO=1. The LTO is disabled by default because it may increase the build time.

Rollback Plan:

Differential Revision: D77899195




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov

desertfire reviewed (approved):
Please fix lint error

helloguo committed:
[AOTI] add flag AOT_INDUCTOR_ENABLE_LTO (#157773)

Summary:

Add env var AOT_INDUCTOR_ENABLE_LTO to enable clang's ThinLTO by setting AOT_INDUCTOR_ENABLE_LTO=1. The LTO is disabled by default because it may increase the build time.

Reviewed By: desertfire

Differential Revision: D77899195
